tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/TestCollectCounterexamplesSeqModelCheckerListener.scala

```diff
@@ -0,0 +1,134 @@
+package at.forsyte.apalache.tla.bmcmt
+
+import at.forsyte.apalache.tla.bmcmt.Checker.Error
+import at.forsyte.apalache.tla.bmcmt.analyses.ExprGradeStoreImpl
+import at.forsyte.apalache.tla.bmcmt.search.ModelCheckerParams
+import at.forsyte.apalache.tla.bmcmt.smt.{RecordingSolverContext, SolverConfig}
+import at.forsyte.apalache.tla.bmcmt.trex.{IncrementalExecutionContext, TransitionExecutorImpl}
+import at.forsyte.apalache.tla.lir.TypedPredefs._
+import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.lir.convenience.tla._
+import org.junit.runner.RunWith
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatestplus.junit.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class TestCollectCounterexamplesModelCheckerListener extends AnyFunSuite {
+
+  private val types = Map("i" -> IntT1(), "b" -> BoolT1())
+
+  private def mkAssign(varName: String, value: Int): TlaEx = {
+    assign(prime(name(varName) ? "i") ? "i", int(value)).typed(types, "b")
+  }
+
+  test("finds cex for invariant violation at initial state") {
+    // construct TLA+ module
+    val initTrans = List(mkAssign("x", 2))
+    val nextTrans = List(mkAssign("x", 2))
+    // Inv == x != 2
+    val notInv = eql(name("x") ? "i", int(2)).typed(types, "b")
+    val inv = not(notInv).typed(types, "b")
+    val module = TlaModule("root", List(TlaVarDecl("x")(Typed(IntT1()))))
+
+    // construct checker input + parameters
+    val checkerInput = new CheckerInput(module, initTrans, nextTrans, None, CheckerInputVC(List((inv, notInv))))
+    val params = new ModelCheckerParams(checkerInput, 0, Map(), false)
+
+    // create utility objects
+    val solver = RecordingSolverContext.createZ3(None, SolverConfig.default)
+    val rewriter = new SymbStateRewriterImpl(solver, new ExprGradeStoreImpl)
+    val ctx = new IncrementalExecutionContext(rewriter)
+    val trex = new TransitionExecutorImpl(params.consts, params.vars, ctx)
+
+    // run the model checker with listener
+    val listener = new CollectCounterexamplesModelCheckerListener()
+    val checker = new SeqModelChecker(params, checkerInput, trex, Seq(listener))
```

Comment:
Since this block seems to repeat for each test, you might want to consider making it a method (parameterized by `params`)

Reply:
Good point, refactored in 32a792353